### ISSUE TITLE
Cannot use Array.prototype.push.apply with NodeList, even though it works properly in all browsers.

### ISSUE THREAD

**Issue by DerGernTod** [microsoft/TypeScript#39312]
**TypeScript Version:**  3.9.x-dev.20200627

**Search Terms:**
strictBindCallApply NodeList

**Code**

```ts
const bodyChildren = document.body.childNodes;
const arr: Node[] = [];
arr.push.apply(arr, bodyChildren);
```

**Expected behavior:**
Doesn't throw errors.

**Actual behavior:**
Throws errors on line 3: Argument of type 'NodeListOf<ChildNode>' is not assignable to parameter of type 'Node[]'.
  Type 'NodeListOf<ChildNode>' is missing the following properties from type 'Node[]': pop, push, concat, join, and 19 more.

**Playground Link:**
[Playground Link](https://www.typescriptlang.org/play?ts=4.0.0-dev.20200627#code/MYewdgzgLgBARiAJgTwMIAsCWAbRAnAUzBgF4ZERgBXAWyKgDoEUHgtcA5JAiAbgFgAUKEiwAhnjwAuGF0QEA2gF1SMZQMES8DAA5UI6BmJ07syABRaANPCRp2+IgEpeQA)

**Related Issues:**
Couldn't find any.

**Comment by MartinJohns** [comment]
Possibly related: #36554
